File: helpers/format.py
from operator import itemgetter


def get_file_string(hashes: dict[str, str], dat: bool):
    name, size, crc, md5, sha1, sha256 = itemgetter(
        'name', 'size', 'crc', 'md5', 'sha1', 'sha256')(hashes)
    return f"<rom name=\"{name}\" size=\"{size}\" crc=\"{crc}\" md5=\"{md5}\" sha1=\"{sha1}\" />\n" if dat else f"""--- {name} ---
Size: {size} bytes
CRC32: {crc}
MD5: {md5}
SHA-1: {sha1}
SHA-256: {sha256}
"""

File: src/format.rs
use crate::hashes::RomInfo;

pub fn format_data(data: &RomInfo, is_dat: bool) -> String {
    match is_dat {
        false => format!(
            "--- {} ---\nSize: {} bytes\nCRC32: {}\nMD5: {}\nSHA-1: {}\nSHA-256: {}\n",
            data.filename, data.size, data.crc32, data.md5, data.sha1, data.sha256
        ),
        true => format!(
            "<rom name=\"{}\" size=\"{}\" crc=\"{}\" md5=\"{}\" sha1=\"{}\" />",
            data.filename, data.size, data.crc32, data.md5, data.sha1
        ),
    }
}
